app/logic/Passkey.scala

```diff
@@ -97,6 +98,36 @@ object Passkey {
       )
     )
 
+  /** Options required by a browser to initiate the authentication process.
+    *
+    * See
+    * [[https://webauthn4j.github.io/webauthn4j/en/#generating-a-webauthn-assertion]].
+    */
+  def authenticationOptions(
+      user: UserIdentity,
+      challenge: Challenge = new DefaultChallenge()
+  ): Try[PublicKeyCredentialRequestOptions] =
+    Try {
+      new PublicKeyCredentialRequestOptions(
+        challenge,
+        null,
+        null,
+        null,
+        null,
+        null
+      )
+    }.recoverWith(exception =>
```

Comment:
I think all these `recoverWith` calls would be simpler with `.adaptError` from Cats, but we can very easily retrofit that approach.

Reply:
Interesting!  There are lots of cats features that I need to learn about.  I'll look at that in a follow-up PR